src/api/app/components/watchlist_component.rb

```diff
@@ -0,0 +1,77 @@
+class WatchlistComponent < ApplicationComponent
+  def initialize(user:)
+    super
+
+    @user = user
+  end
+
+  def current_page_item_in_watchlist?
+    @user.watched_items.includes(:watchable).where(watchable: current_page_item).any?
+  end
+
+  def current_page_item_watchable?
+    params[:package] || params[:project] || params[:number]
```

Comment:
Where is `params` coming from in the context of a view component?

Reply:
`params` is simply accessible from the view component. But now that you mention this, maybe it's better to pass what we need to the view component initializer and don't rely on `params` inside the view component... Let me see if that is doable.

What do you think?

Reply:
> it's better to pass what we need to the view component initializer

Yes, I agree. We should avoid global state in view components, it makes them difficult to test. I'll add a note about this in our [developer page for view components](https://github.com/openSUSE/open-build-service/wiki/View-Components).

Reply:
It's not available from the view component specs, so we need to change this...

Reply:
@dmarcoux could you have another look? I added a new commit refactoring this part of the code.

Reply:
We finally agreed on passing the parameters like this `WatchlistComponent.new(user: User.session!, project_name: params[:project], package_name: params[:package], bs_request_number: params[:number])` so we don't use `params` inside the view component.